SE-XXX Typed throws.md

```diff
@@ -477,7 +520,11 @@ do {
 }
 ```
 
-Type inference would benefit also of typed `throws` like shown in the following examples:
+### Type inference
+
+Type inference is a powerful tool built-in Swift language that leverages many boilerplate into simplier and easier to read code. This rules apply to the current proposal, specifically regarding `enum` errors.

```

Comment:
> Type inference is a powerful tool built-in Swift language that leverages many boilerplate into simplier and easier to read code.

everyone reading this should know what type inference is 😅

Reply:
It was to give an intro to enums, but how'd you write that paragraph

Reply:
As a proposal draft I don't think that we need prose that does not contain new information. I would remove the paragraph